packages/strapi-plugin-content-type-builder/admin/src/containers/Form/index.js

```diff
@@ -514,9 +526,12 @@ export class Form extends React.Component { // eslint-disable-line react/prefer-
   }
 
   renderModalBodyChooseAttributes = () => {
-    const attributesDisplay = has(this.context.plugins.toJS(), 'upload') 
-      ? forms.attributesDisplay.items
-      : forms.attributesDisplay.items.filter(obj => obj.type !== 'media'); // Don't display the media field if the upload plugin isn't installed
+    const attributesDisplay = forms.attributesDisplay.items;
+
+    // Don't display the media field if the upload plugin isn't installed
+    if (!has(this.context.plugins.toJS(), 'upload')) {
+      attributesDisplay.splice(8, 1);
```

Comment:
You might need to clone the `attributesDisplay ` before splicing it. I remember that I used to do the same logic and it caused the attributesForm to be buggy.

```
const attributesDisplay = (() => {
  const list = forms.attributesDisplay.items;

  if (!this.context.plugins.toJS().hasOwnProperty('upload')) {
     return list.filter(obj => obj.type !== 'media';
  }
   return list;
})();
```

Reply:
Same for here,  it shouldn't be update, i put code back here too